client/starwhale/api/_impl/dataset/model.py

```diff
@@ -998,3 +998,51 @@ def dataset(
         )
 
         return ds
+
+    @classmethod
+    def from_dict(
+        cls, name: str = "", data: dict = {}, field_selector: str = ""
+    ) -> Dataset:
+        """Create a new dataset from a dict.
+        Arguments:
+            name: (str, required) The dataset name you would like to use.
+            data: (dict, required) The dict from which you would like to create this dataset
+            field_selector: (str, optional) The filed from which you would like to extract dataset array items.
+                The default value is "" which indicates that the dict is an array contains all the items.
+            Returns:
+                A Dataset Object
+            Examples:
+            ```python
+            from starwhale import Dataset
+            myds = Dataset.from_dict("translation", [{"en":"hello","zh-cn":"你好"},{"en":"how are you","zh-cn":"最近怎么样"}])
+            print(myds[0].features.en)
+            ```
+            ```python
+            from starwhale import Dataset
+            myds = Dataset.from_dict("translation", {"content":{"child_content":[{"en":"hello","zh-cn":"你好"},{"en":"how are you","zh-cn":"最近怎么样"}]}},"content.child_content")
+            print(myds[0].features["zh-cn"])
+            ```
```

Comment:
add a example for `field_selector`?

Reply:
https://github.com/star-whale/starwhale/blob/ff6a81c8ab88bc8ebb0c79e49841dbd9844ef563/client/starwhale/api/_impl/dataset/model.py#L1021
"content.child_content" is an example